src/app/modules/group/http-services/remove-group.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { forkJoin, Observable, of } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { appConfig } from '../../../shared/helpers/config';
+import { SimpleActionResponse } from '../../../shared/http-services/action-response';
+import { catchError, map } from 'rxjs/operators';
+
+export function parseResults(data: Map<string, SimpleActionResponse>): { countRequests: number, countSuccess: number, errorText?: string } {
+  const values = Array.from(data.values());
+
+  return {
+    countRequests: data.size,
+    countSuccess: values.map<number>(state => (state.success ? 1 : 0)).reduce((acc, res) => acc + res, 0),
+    errorText: values.some(value => !!value.error_text) ? $localize`The group(s) must be empty` : undefined,
+  };
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RemoveGroupService {
+  constructor(private http: HttpClient) {
+  }
+
+  remove(id: string): Observable<SimpleActionResponse> {
+    return this.http.delete<SimpleActionResponse>(`${appConfig.apiUrl}/groups/${id}`);
+  }
+
+  removeBatch(ids: string[]): Observable<{ countRequests: number, countSuccess: number, additionalMessage?: string }> {
+    return forkJoin(
+      ids.map(id => this.remove(id).pipe(catchError(({ error }) => of(error))))
+    ).pipe(
+      map(data => parseResults(new Map(Object.entries(data)))),
```

Comment:
Here `data` is an array right ?
That means a Map is created from an array entries and provided to `parseReults()`, which does a `Array.from(theMap.values())`

Why not passing the array directly ?

Reply:
This logic partly based on "as done in pending request component.", and I guess it was done for easiest way to get just values for reduce total count. Do you want I'm improve it?

Reply:
I checked it out:
The `group-users` service returns the map, not the `parseResults()` result. This way it _may_ make sense because the subscriber might use the map in another way.

Here, the service returns directly the `parseResults()` result and not the map, so there's no chance this map is going to be used anywhere in any subscriber, and it's a same in the `remove-subgroups` service

To me it _might_ make sense in the `group-users` service but not here nor in `remove-subgroup` service, I think we should change those `remove-groups` and `remove-subgroups` services only